catalyst-gateway/bin/src/service/common/types/cardano/query/cat_id_or_stake.rs

```diff
@@ -0,0 +1,125 @@
+//! Query Parameter that can take a CIP-19 stake address or a Catalyst Id.
+//!
+//! Allows us to have one parameter that can represent two things, uniformly.
+
+// cSpell:ignoreRegExp cardano/Fftx
+
+use std::sync::LazyLock;
+
+use const_format::concatcp;
+use poem_openapi::{
+    registry::{MetaSchema, MetaSchemaRef},
+    types::{ParseFromParameter, ParseResult, Type},
+};
+use serde_json::Value;
+
+use crate::service::common::types::cardano::{
+    catalyst_id::{self, CatalystId},
+    cip19_stake_address::{self, Cip19StakeAddress},
+};
+
+/// A Query Parameter that can take a CIP-19 stake address, or a Catalyst Id
+/// identifier.. Defining these are mutually exclusive, as a single parameter is required
+/// to be used.
+#[derive(Clone)]
+#[allow(clippy::large_enum_variant, dead_code)]
+pub(crate) enum CatIdOrStake {
+    /// A CIP-19 stake address
+    Address(Cip19StakeAddress),
+    /// A catalyst id
+    CatId(CatalystId),
+}
+
+impl TryFrom<&str> for CatIdOrStake {
+    type Error = anyhow::Error;
+
+    fn try_from(value: &str) -> std::result::Result<Self, Self::Error> {
+        if let Ok(cat_id) = CatalystId::try_from(value) {
+            Ok(Self::CatId(cat_id))
+        } else if let Ok(stake_addr) = Cip19StakeAddress::try_from(value) {
+            Ok(Self::Address(stake_addr))
+        } else {
+            anyhow::bail!("Not a valid \"Catalyst Id or Stake Address\" parameter.");
+        }
+    }
+}
+
+/// Title.
+const TITLE: &str = "Catalyst Id or Stake Address.";
+/// Description.
+const DESCRIPTION: &str = "Restrict the query to this Catalyst Id or Stake Address .
+If neither are defined, the stake address(es) from the auth tokens role0 registration are used.";
```

Comment:
This is somewhat wrong, we are using the stake address from auth token, but we are using that auth token which is a cat-id
Maybe change to
`If neither are defined, the auth token is used.";`
There is one auth token right? so it should be singular.